### PR TITLE
resolve cycling issue

### LINKED ISSUE #42: integer solutions cycling
So Miles and I just found that the integer solutions are cycling when we run compare.jl in the dropbox with node problem (and also the full graph problem).

We printed them using the following code added to line 988 of conic.jl
   ```
mip_solution = getValue(m.mip_x)
int_indices = [getCategory(Variable(mip_model,i)) != :Cont for i in 1:MathProgBase.numvar(mip_model)]
@show mip_solution[int_indices]
```

This was the output when we set the random seed to 100
> Best integer =   7.257441e-01  Node =       0  Best node =   7.529509e-04
> mip_solution[int_indices] = [1.0,0.0,1.0,0.0,1.0,0.0,0.0,1.0,0.0]

### PR DESCRIPTION
**Pull request by emreyamangil**
This PR resolves cycling issue.

Closes #42

### CODE REVIEW

**Comment by mlubin** [comment]
Tests are failing.

**Comment by mlubin** [comment]
Problems that were optimal before are now suboptimal. We should check the optimality conditions before deciding that the problem is suboptimal.

**Comment by emreyamangil** [comment]
Woops! Updating asap!

**Comment by emreyamangil** [comment]
Tests are passing on my machine after this fix!

**Comment by chriscoey** [comment]
woohoo!!! good job

File: src/conic.jl [hunk 1]
```diff
@@ -975,13 +981,16 @@ function MathProgBase.optimize!(m::PajaritoConicModel)
...
                 (m.verbose > 1) && println("CPUTIME: $(time() - start)")
                 (m.verbose > 1) && println("Number of OA iterations: $iter")
+            else #cycle_indicator
+                m.status = :Suboptimal
             end
+            m.iterations = iter
             #break
         end
         (m.verbose > 0) && (m.algorithm == "OA") && @printf "%9d   %+.7e   %+.7e   %+.7e   %+.7e   %+.7e   %+.7e\n" iter mip_objval conic_objval optimality_gap m.objval primal_infeasibility OA_infeasibility
+        (cycle_indicator) && warn("Mixed-integer cycling detected, terminating Pajarito...")
```
Comment (mlubin) [inline src/conic.jl]: The condition should be ``(cycle_indicator && m.status != :Optimal)``. Right now it prints the warning on every solution!

Comment (emreyamangil) [inline src/conic.jl]: Confusion! But why?? If we separate cycle_indicator should be false!

Comment (mlubin) [inline src/conic.jl]: This is when we terminate, not on every iteration.

Comment (emreyamangil) [inline src/conic.jl]: Oh this is happening when the dcp_model was feasible?

Comment (mlubin) [inline src/conic.jl]: This is happening when ``cyle_indicator == true`` and we satisfy optimality

**Pull request merged**